packages/wonder-blocks-layout/components/media-layout.test.js

```diff
@@ -0,0 +1,343 @@
+                            color: "blue",
+                        },
+                    }),
+                    small: StyleSheet.create({
+                        test: {
+                            color: "orange",
+                        },
+                    }),
+                    medium: StyleSheet.create({
+                        test: {
+                            color: "teal",
+                        },
+                    }),
+                };
+                resizeWindow(size);
+
+                // Act
+                const style = getStyle(styleSheets);
+                const color = {
+                    small: "orange",
+                    medium: "teal",
+                }[size];
+
+                // Assert
+                expect(style.color).toEqual(color);
+            });
+        }
+
+        for (const size of ["medium", "large"]) {
+            it(`"${size}" styles should win over "mdOrLarger" styles`, async () => {
+                // Arrange
+                const styleSheets = {
+                    mdOrLarger: StyleSheet.create({
+                        test: {
+                            color: "blue",
+                        },
+                    }),
+                    medium: StyleSheet.create({
+                        test: {
+                            color: "teal",
+                        },
+                    }),
+                    large: StyleSheet.create({
+                        test: {
+                            color: "magenta",
+                        },
+                    }),
+                };
+                resizeWindow(size);
+
+                // Act
+                const style = getStyle(styleSheets);
+                const color = {
+                    medium: "teal",
+                    large: "magenta",
+                }[size];
+
+                // Assert
+                expect(style.color).toEqual(color);
+            });
+        }
+    });
+
+    describe("window resizing", () => {
+        it("should update the style when the window gets smaller", async () => {
+            // Arrange
+            const styleSheets = {
+                large: StyleSheet.create({
+                    test: {
+                        color: "blue",
+                    },
+                }),
+                small: StyleSheet.create({
+                    test: {
+                        color: "orange",
+                    },
+                }),
+            };
+
+            resizeWindow("large");
+
+            const wrapper = mount(
+                <MediaLayout styleSheets={styleSheets}>
+                    {({mediaSize, mediaSpec, styles}) => {
+                        return <View style={styles.test}>Hello, world!</View>;
+                    }}
+                </MediaLayout>,
+            );
+
+            resizeWindow("small");
+            wrapper.update();
+
+            // Act
+            const style = wrapper.find("div").prop("style");
```

Comment:
I think some of the things in the `Arrange` part should be here as they are the action that is being tested. This line specifically is really a sampling point after the action (it belongs here, but it is not actually the action under test).

Reply:
Still think this is a valid point. There's no actual action happening on this line. I think the second resize and update call should be here too; they're what trigger the code that we're testing, right?